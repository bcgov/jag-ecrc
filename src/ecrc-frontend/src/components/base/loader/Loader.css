.page-loader {
  height: 40px;
  width: 40px;
  color: #494949;
  border: 5px solid;
  border-radius: 50%;
  border-top-color: transparent;
  animation: rotate 2s linear infinite;
  display: inline-block;
}

.btn-loader {
  height: 15px;
  width: 15px;
  color: #ffffff;
  border: 2px solid;
  border-radius: 50%;
  border-top-color: transparent;
  animation: rotate 2s linear infinite;
  display: inline-block;
  margin-left: 10px;
  margin-top: 3px;
  vertical-align: text-top;
}

@keyframes rotate {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
