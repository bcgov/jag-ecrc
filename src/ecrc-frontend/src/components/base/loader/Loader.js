import React from "react";
import PropTypes from "prop-types";
import "./Loader.css";

export default function Loader({ page }) {
  let loaderCss = "btn-loader";
  if (page) loaderCss = "page-loader";
  return <div className={loaderCss} />;
}

Loader.propTypes = {
  page: PropTypes.bool
};

Loader.defaultProps = {
  page: false
};
