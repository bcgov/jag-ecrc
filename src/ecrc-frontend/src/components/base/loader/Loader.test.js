import React from "react";
import { create } from "react-test-renderer";

import Loader from "./Loader";

describe("Loader Component", () => {
  test("Page loader matches the snapshot", () => {
    const page = true;
    const loader = create(<Loader page={page} />);
    expect(loader.toJSON()).toMatchSnapshot();
  });

  test("Btn loader matches the snapshot", () => {
    const page = false;
    const loader = create(<Loader page={page} />);
    expect(loader.toJSON()).toMatchSnapshot();
  });
});
