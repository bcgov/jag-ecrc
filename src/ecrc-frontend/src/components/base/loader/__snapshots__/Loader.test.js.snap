// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Matches the snapshot 1`] = `
<div
  className="btn-loader"
/>
`;
